probes/hasDangerousWorkflowScriptInjection/impl.go

```diff
@@ -48,6 +50,7 @@ func Run(raw *checker.RawResults) ([]finding.Finding, string, error) {
 	}
 
 	var findings []finding.Finding
+	contents := make(map[string][]byte)
```

Comment:
This can be simplified, since the findings are always grouped by workflow (all the findings for `foo.yml`, then all the findings for `bar.yml`, etc). We therefore don't need to keep a map of all the contents, but simply store the contents of the "current" workflow; when we start on a new workflow, we replace `contents` with the contents of the new workflow.

Reply:
This is because `r.Workflows` is populated by [`checkJobForUntrustedCodeCheckout`](https://github.com/ossf/scorecard/blob/26ffe63a3e282f55c7d685e56fa22b102196cce6/checks/raw/dangerous_workflow.go#L196) and [`checkJobForUntrustedCodeCheckout`](https://github.com/ossf/scorecard/blob/26ffe63a3e282f55c7d685e56fa22b102196cce6/checks/raw/dangerous_workflow.go#L256), which are called by `validateUntrustedCodeCheckout` and `validateScriptInjection`, respectively, both of which are called by `validateGitHubActionWorkflowPatterns`, which itself is applied to each workflow file, in order, by [`DangerousWorkflow`](https://github.com/ossf/scorecard/blob/26ffe63a3e282f55c7d685e56fa22b102196cce6/checks/raw/dangerous_workflow.go#L68) (using [`OnMatchingFileContentDo`](https://github.com/ossf/scorecard/blob/main/checks/fileparser/listing.go#L76).

Reply:
Hum yeah it makes sense. I'll change that and at least we will consume less memory on doing it this way

Reply:
Done!